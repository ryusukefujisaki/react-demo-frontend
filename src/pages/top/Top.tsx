import PageMovingCounter from '@/components/PageMovingCounter'

function Top(): JSX.Element {
  return (
    <div className="min-h-screen my-0 w-max mx-auto">
      <div className="text-3xl py-12">
        Top
      </div>
      <PageMovingCounter />
    </div>
  )
}

export default Top
