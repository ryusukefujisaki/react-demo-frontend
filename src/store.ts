import { configureStore } from '@reduxjs/toolkit'
import pageMovingCounterReducer from '@/features/pageMovingCounter/pageMovingCounterSlice'

const store = configureStore({
  reducer: {
    pageMovingCounter: pageMovingCounterReducer
  }
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

export default store
