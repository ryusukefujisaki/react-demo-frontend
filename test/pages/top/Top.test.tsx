import * as renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import Top from '@/pages/top/Top'
import store from '@/store'

it('snapshot test', () => {
  const page = renderer.create(
    <Provider store={store}>
      <Top />
    </Provider>
  )
  let tree = page.toJSON()
  expect(tree).toMatchSnapshot()
})
