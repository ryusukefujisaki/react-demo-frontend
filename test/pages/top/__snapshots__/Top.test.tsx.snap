// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="min-h-screen my-0 w-max mx-auto"
>
  <div
    className="text-3xl py-12"
  >
    Top
  </div>
  <table
    className="border-collapse py-3"
  >
    <thead>
      <tr>
        <th
          className="border border-slate-300 p-3"
          colSpan={2}
        >
          <img
            alt="Redux logo"
            className="logo redux h-4 p-0"
            src="/redux.svg"
          />
          Redux
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          className="border border-slate-300 p-3 bg-slate-100"
        >
          Page Moving Count
        </th>
        <td
          className="border border-slate-300 p-3"
        >
          1
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
